mlrun/model_monitoring/db/stores/sqldb/sql_store.py

```diff
@@ -441,14 +439,15 @@ def update_last_analyzed(
         """
         self._init_monitoring_schedules_table()
 
-        application_filter_dict = self.filter_endpoint_and_application_name(
-            endpoint_id=endpoint_id, application_name=application_name
-        )
+        criteria = [
+            self.MonitoringSchedulesTable.endpoint_id == endpoint_id,
+            self.MonitoringSchedulesTable.application_name == application_name,
+        ]
```

Comment:
You are using this lines or part of them for all the tables a lot of times in your code, I think you can consider to create some helper method such that `_get_criteria(table_name, endpoint_id, application_name)` and to use it in all the relevant places for all the tables.

Reply:
Done in 90b4988